### PR TITLE
feat: detach font icons from the themes

### PR DESCRIPTION
**Pull request by inikolova**
BREAKING CHANGE: detach font icons from the themes

In this PR:

- Remove themes dependencies to the @progress/kendo-font-icons package;
- Add themes dependencies to the @progress/kendo-svg-icons package;
- Include svg icons styles in packages/{theme}/scss/icons/_layout.scss files;
- Add a **k-font-icon** class for all font icons;
- Add a **k-icon** class for all svg icons;

Part of https://github.com/telerik/kendo-themes-private/issues/124

**Note**: Should be merged after https://github.com/telerik/kendo-icons/pull/186

### CODE REVIEW

**Review by joneff — APPROVED** [review]
Now that's a long overdue change!

File: packages/html/src/icon/tests/icons.tsx [hunk 1]
```diff
@@ -615,6 +614,7 @@ const styles = `
 export default () =>(
     <>
         <style>{styles}</style>
+        <link rel="stylesheet" href="https://unpkg.com/@progress/kendo-font-icons@1.9.0/dist/index.css" />
```
Comment (Juveniel) [inline packages/html/src/icon/tests/icons.tsx:617]: This is not reliable. I would rather remove the test for the font-icons.

File: packages/default/scss/button/_layout.scss [hunk 2]
```diff
@@ -93,7 +93,7 @@
                 align-items: center;
                 justify-content: center;
 
-                &.k-svg-icon > svg,
+                > svg,
```
Comment (Juveniel) [inline packages/default/scss/button/_layout.scss:96]: we could remove the > svg selector here

File: packages/default/scss/badge/_layout.scss [hunk 3]
```diff
@@ -46,7 +46,7 @@
         max-height: 1em;
         font-size: inherit;
     }
-    .k-badge-icon.k-svg-icon svg {
+    .k-badge-icon svg {
```
Comment (Juveniel) [inline packages/default/scss/badge/_layout.scss:49]: we probably need the old selector here

File: packages/default/scss/grid/_layout.scss [hunk 4]
```diff
@@ -300,18 +298,6 @@
...
-                padding-block: $kendo-grid-cell-padding-y;
-                padding-inline: 0;
-                width: 100%;
-                height: 100%;
-                // That causes an issue with grid material. Commenting it for now
-                // -- joneff
-                // line-height: $kendo-line-height-md;
-                display: inline-block;
-                outline: 0;
-            }
```
Comment (Juveniel) [inline packages/default/scss/grid/_layout.scss:314]: we should not remove this

File: packages/fluent/scss/input/_layout.scss [hunk 5]
```diff
@@ -241,7 +239,6 @@
     .k-input-icon,
     .k-input-validation-icon,
     .k-input-loading-icon {
-        &.k-svg-icon > svg,
```
Comment (Juveniel) [inline packages/fluent/scss/input/_layout.scss:244]: we need this selector here

File: packages/default/scss/input/_layout.scss [hunk 6]
```diff
@@ -233,7 +231,6 @@
     .k-input-icon,
     .k-input-validation-icon,
     .k-input-loading-icon {
-        &.k-svg-icon > svg,
```
Comment (Juveniel) [inline packages/default/scss/input/_layout.scss:236]: this selector is needed - check visual tests

File: tests/_output/bootstrap/loading/loading.png
Comment (Juveniel) [inline tests/_output/bootstrap/loading/loading.png:1]: icon misaligned here - check test

File: tests/_output/bootstrap/scrollview/scrollview.png
Comment (Juveniel) [inline tests/_output/bootstrap/scrollview/scrollview.png:1]: check test

File: tests/_output/bootstrap/pivotgrid/pivotgrid-settings-panel-reordering.png
Comment (Juveniel) [inline tests/_output/bootstrap/pivotgrid/pivotgrid-settings-panel-reordering.png:1]: check test

File: packages/default/scss/button/_layout.scss (hunk 2 above)
Comment (inikolova) [reply packages/default/scss/button/_layout.scss:96]: We need this selector, otherwise the icons increases its width/height

**Review by Juveniel — APPROVED** [review]

**Pull request merged**